tfe/resource_tfe_variable.go

```diff
@@ -15,6 +15,9 @@ func resourceTFEVariable() *schema.Resource {
 		Read:   resourceTFEVariableRead,
 		Update: resourceTFEVariableUpdate,
 		Delete: resourceTFEVariableDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
```

Comment:
Well.. I don't know how this could have ever worked for you. I cannot get this to work. Not even with the totally wierd ID format you tried `"hashicorp-v2/canary|var-aBTYFA56FS"`.

But like we already discussed, we should just add support for the variable read endpoint to `go-tfe` and then this will be fixed. But I first want to merge this PR before I update the vendored `go-tfe`, so I suggest we test this one again after I updated the package (which will be before we release the next version of course).

Reply:
Ah! I was actually messing with an importer for that and might have had it in for my final testing. Here's my code:

```

func resourceTFEVariableImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
	workspaceID := ""
	key := d.Id()

	if strings.Contains(d.Id(), "|") {
		parts := strings.SplitN(d.Id(), "|", 2)
		if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
			return nil, fmt.Errorf("ID must have format <organization>/<workspace>|<id>")
		}
		workspaceID, key = parts[0], parts[1]
	}

	d.Set("workspace_id", workspaceID)
	d.SetId(key)

	if err := resourceTFEVariableRead(d, meta); err != nil {
		return nil, err
	}

	return []*schema.ResourceData{d}, nil
}
```